src/dmd/func.d

```diff
@@ -2496,6 +2496,55 @@ extern (C++) class FuncDeclaration : Declaration
             error("parameters must be `main()` or `main(string[] args)`");
     }
 
+    /***********************************************
+     * Check all return statements for a function to verify that returning
+     * using NRVO is possible.
+     *
+     * Returns:
+     *      true if the result cannot be returned by hidden reference.
```

Comment:
Returning `true` if NRVO is **not** possible feels kinda wrong

Reply:
A lot of the `checkFoo` functions do that. I don't think thats too out of place.

Reply:
Fair enough